src/main/groovy/net/wasdev/wlp/gradle/plugins/utils/ServerConfigDocument.java

```diff
@@ -188,37 +203,69 @@ private static void parseInclude(Document doc) throws Exception {
             }
         }
     }
-    
+
+    //Checks for application names in the document. Will add locations without names to a Set
+    private static void parseNames(Document doc, String expression) throws Exception {
+        // parse input document
+        XPath xPath = XPathFactory.newInstance().newXPath();
+        NodeList nodeList = (NodeList) xPath.compile(expression).evaluate(doc, XPathConstants.NODESET);
```

Comment:
see if you can use a regular expression to get application and webApplication names in one call.